routing/rt/graph/graph.go

```diff
@@ -0,0 +1,536 @@
+// Copyright (c) 2016 Bitfury Group Limited
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php
+
+package graph
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+	"os"
+	"strconv"
+	"bytes"
+)
+
+var (
+	CyclicDataError         = errors.New("Cyclic data")
+	PathNotFoundError       = errors.New("Path not found")
+	NodeNotFoundError       = errors.New("Node not found")
+	EdgeNotFoundError       = errors.New("Edge not found")
+)
+
+type ID struct {
```

Comment:
Vertex seems good to me. 
Collapsing will make it impossible to distinguish [33]byte from ID in type switch for example.